weave/trace/weave_client.py

```diff
@@ -1572,4 +1577,60 @@ def sanitize_object_name(name: str) -> str:
     return res
 
 
+def convert_paths_to_images(obj: str) -> typing.Union[str, PathImage]:
+    """Load or download paths to images as PathImage objects.
+
+    If the path is a local image, open it and return a PathImage object.
+    If the path is a remote image, download it and return a PathImage object.
+    """
+    if is_remote_image(obj) or is_local_image(obj):
+        return PathImage(path=obj, data=None)
+    elif is_base64_image(obj):
+        return PathImage(data=obj, path=None)
+
+    return obj
+
+
+def parse_obj(obj: Any, parsers: list[Callable[..., Any]]) -> Any:
+    """Parse an object with conversion parsers.
+    Accepts a list of functions that do operations on the object.
+        parser functions must include type hints, and accept exactly
+        one parameter called "obj" that can be modified and returned
+        example:
+            def atoi(obj: str):
+                return int(obj)
+
+
+    Returns the modified object
+    """
+    # Dont mutate reffed objects
+    if get_ref(obj):
+        return obj
+
+    # recurse through container types
+    if isinstance(obj, dict):
+        # Do any dict-type parsing
+        for parser in parsers:
+            param_types = typing.get_type_hints(parser)
```

Comment:
This is probably ok, but worth double checking because the behaviour of `get_type_hints` can change between python versions: https://docs.python.org/3/library/typing.html#typing.get_type_hints